### PR TITLE
Refactor: Azure Stack HCI Cluster

### PR DESCRIPTION
**Pull request by neil-yechenwei**
This PR is for updating code against the comments from https://github.com/terraform-providers/terraform-provider-azurerm/pull/9134#pullrequestreview-547894458.

--- PASS: TestAccStackHCICluster_basic (141.65s)
--- PASS: TestAccStackHCICluster_complete (142.16s)
--- PASS: TestAccStackHCICluster_requiresImport (162.70s)
--- PASS: TestAccStackHCICluster_update (200.45s)

### CODE REVIEW

**Comment by neil-yechenwei** [comment]
@tombuildsstuff, I assume this PR has fixed all comments from #9134 (review). Please have a look. Thanks.

File: website/docs/r/stack_hci_cluster.html.markdown [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+* `resource_group_name` - (Required) The name of the Resource Group where the Azure Stack HCI Cluster should exist. Changing this forces a new resource to be created.
+
+* `location` - (Required) The Azure Region where the Azure Stack HCI Cluster should exist. Changing this forces a new resource to be created.
+
+* `client_id` - (Required) The Client ID of the Azure Active Directory which is used by the Azure Stack HCI Cluster. Changing this forces a new resource to be created.
+
+* `tenant_id` - (Optional) The Tenant ID of the Azure Active Directory which is used by the Azure Stack HCI Cluster. Changing this forces a new resource to be created.
+
+~> **NOTE** By default, the `tenant_id` would be sourced from the `ARM_TENANT_ID` Environment Variable.
```
Comment (tombuildsstuff) [inline website/docs/r/stack_hci_cluster.html.markdown:50]: Not all users use service principals, so this wants to be:

```suggestion
~> **NOTE** If unspecified the Tenant ID of the Provider will be used.
```

**Review by tombuildsstuff — APPROVED** [review]
hi @neil-yechenwei 

Thanks for this PR - taking a look through there's one minor comment but this otherwise LGTM 👍 

Thanks!

**Comment by tombuildsstuff** [comment]
Tests pass:

Screenshot 2020-12-10 at 12 52 10

**Comment by tombuildsstuff** [comment]
The generation failures are unrelated and will be fixed via #9802

**Pull request merged**